### ISSUE TITLE
Server Error

### ISSUE THREAD

**Issue by seaneady** [dwyl/mvp#281]
![image](https://user-images.githubusercontent.com/122034315/214088901-e882abcd-4abe-429c-9466-cfa994dab7f5.png)
Reporting a bug which I encountered when attempting to sign up to the mobile and web version of the MVP.

**Comment by LuchoTurtle** [comment]
🤔 

Did this error occur when you logged in? I've tried logging in from an anonymous window and it seems to be working properly. 

<img width="1000" alt="image" src="https://user-images.githubusercontent.com/17494745/214098218-30cf2319-6a83-462c-b747-b5321508e675.png">


What steps did you do to get to this error?

**Comment by LuchoTurtle** [comment]
Hm, okay, when pushing changes to https://github.com/dwyl/mvp/pull/280, it seems the CI errors when deploying it to Fly.io.

https://github.com/dwyl/mvp/actions/runs/3988547375/jobs/6839852547

This might be related.  Let me see if I can figure out what's going on 👍

**Comment by LuchoTurtle** [comment]
Uhm. I re-ran that job and it seems to be working now.
https://github.com/dwyl/mvp/actions/runs/3988547375

Perhaps some networking issues on Fly.io's side?
Logs don't show anything out of the ordinary either -> https://fly.io/apps/mvp-pr-280/monitoring

Is your issue still occurring @seaneady ? Everything seems fine on my end 😄

**Comment by seaneady** [comment]
<img width="1440" alt="Screenshot 2023-01-23 at 5 32 12 PM" src="https://user-images.githubusercontent.com/122034315/214108181-591479a4-ae3c-46dc-8983-82269d996449.png">

I tried it in two different browsers and still getting this error.

**Comment by seaneady** [comment]
also tried it on other computers too and I got the same thing.

**Comment by SimonLab** [comment]
I can see the following error on the logs, not sure yet if it's related:
![image](https://user-images.githubusercontent.com/6057298/214109197-5c1b5cc1-b5e4-454d-8e06-7426cf574849.png)

**Comment by LuchoTurtle** [comment]
Database logs show this. https://fly.io/apps/mvp-db/monitoring

```
2023-01-23T17:26:07.005 app[12a3726f] lhr [info] keeper | 2023-01-23 17:26:07.002 UTC [32321] DETAIL: Key (person_id)=(209) is not present in table "people".

2023-01-23T17:26:08.021 app[12a3726f] lhr [info] keeper | 2023-01-23 17:26:08.021 UTC [32321] ERROR: insert or update on table "items" violates foreign key constraint "items_person_id_fkey"
```

This is probably the issue, perhaps Sean's person row was deleted from the database for whatever reason and can't properly login (?). 

I don't really know how to access the database and make changes to it, going to leave it to the pros. @nelsonic @SimonLab

**Comment by SimonLab** [comment]
From https://github.com/dwyl/mvp/issues/281#issuecomment-1400719453 I think maybe an error has been created on Fly.io side. the `auth.dwyl.com` is accessible for me but it makes sense that the queries won't work on the mvp if the `person_id` is not retrieved from the `auth` app.

**Comment by nelsonic** [comment]
@seaneady thanks very much for reporting this. 👌 (😢)
I strongly suspect that this is a browser cookies issue. 🍪 
Specifically that something went wrong in the auth process and the `JWT` is corrupt/outdated. 🙄 
Please confirm which web browser you are using so we can help with clearing. 🙏

**Comment by seaneady** [comment]
No problem. I tried from Chrome and Safari on two separate computers and got the same error. Could it have something to do with the fact that I am trying to log in from South Africa maybe?

**Comment by nelsonic** [comment]
It's possible that `CloudFlare` is blocking the requests. Super lame if they are! 😢 
Might need to fire up `VPN` to see if we can reproduce the error. 💭

**Comment by nelsonic** [comment]
@seaneady when you're back at your desk, please LMK.
We can remote-pair on debugging this.
Continue: https://fly.io/apps/authprod/monitoring

**Comment by seaneady** [comment]
@LuchoTurtle sorry I'm only getting back to you now. Been a bit hectic on my side today. Would 3pm your time work? I have another call starting now 2pm(your time).

**Comment by LuchoTurtle** [comment]
@seaneady sorry, just saw this! 
I'm not sure I'll help much "unblocking this", as I don't have much experience with Fly.io as @nelsonic or @SimonLab and I don't want to "break" anything.
But if Nelson pairs with you on this, I can surely tag along!

**Comment by nelsonic** [comment]
Let's look at this immediately after standup today (stay on the Zoom call). 👌 
We need @seaneady's help to debug because only he is seeing the error. 🙈

**Comment by nelsonic** [comment]
We're taking a look at this _now_ on the Zoom call. 🔥

**Comment by nelsonic** [comment]
This is going to be a bit of a mission. We need to understand the Database constraint error as shared by @LuchoTurtle above ⏫  https://github.com/dwyl/mvp/issues/281#issuecomment-1400821083

**Comment by SimonLab** [comment]
I think the issue is that we still have a foreign key constraint on the `items` table for people:
![image](https://user-images.githubusercontent.com/6057298/214548836-9e3daed8-4d88-4201-867d-021fde619664.png)

So if we remove the `items_person_id` foreign key it might resolve the issue

**Comment by SimonLab** [comment]
We can do this directly on postgres with the command:
```sql
ALTER TABLE table_name 
   DROP CONSTRAINT foreign_key_name; 
```

What I'm not sure is why the constraint stayed when we removed it from the migrations. Maybe there is an migration command to do this directly in the Phoenix project

**Comment by SimonLab** [comment]
On my localhost, after running `mix ecto.reset` I have the following:
![image](https://user-images.githubusercontent.com/6057298/214552441-55d2c665-7c9a-4f9c-a773-eb0f1903e3b4.png)

We can see that I don't have the foreign key defined.
Maybe there was an issue running a migration on fly, I think the easier/quickest way to fix this issue is to run the postgres command directly on fly (using `flyctl postgres connect` command).

**Comment by nelsonic** [comment]
Ok. we have a snapshot created: https://github.com/dwyl/learn-devops/issues/87
I'm going to try and attach the `mvp` App to the new cluster: `debugging-281`
and see if we can run this `ALTER TABLE` command on it. 👌

**Comment by nelsonic** [comment]
Applied the `ALTER TABLE` command on `mvp-db` and it should no longer have the constraint error ...
@seaneady please test when you can. 🙏

**Comment by SimonLab** [comment]
The people table has been removed from the phoenix app without creating a migration. https://github.com/dwyl/mvp/commit/b7d7aedfbc89e647ec944b33bd27e2c725e89fb5
So on Fly we still have the people table and the foreign keys linked to this table.
On localhost when `mix ecto.reset` is run the database is ok, because the migration file for people has been deleted.

**Comment by nelsonic** [comment]
Yeah, that's my bad. Thought it had been removed cleanly. Thanks for investigating @SimonLab 🔍 🕵️

**Comment by LuchoTurtle** [comment]
> The people table has been removed from the phoenix app without creating a migration. [b7d7aed](https://github.com/dwyl/mvp/commit/b7d7aedfbc89e647ec944b33bd27e2c725e89fb5) So on Fly we still have the people table and the foreign keys linked to this table. On localhost when `mix ecto.reset` is run the database is ok, because the migration file for people has been deleted.

But if the migration file has been deleted from the repo, doesn't Fly.io just run the migrations from scratch? It shouldn't create the `people` table if no migration file is present in the repo 🤔

**Comment by nelsonic** [comment]
Nope. Sadly, that's now how it works. I only runs the _new_ migrations. The old ones are already executed.
This is both a _benefit_ and a downside of migrations.

**Comment by nelsonic** [comment]
Ok. we need @seaneady to re-test on his side ... 
meanwhile I can confirm that everything on the `MVP` still works for me. 👌

**Comment by seaneady** [comment]
Tested and can confirm that I can login via Github however when I attempt to login with my email address it still gives me an error message as per screenshot.
<img width="1111" alt="Screenshot 2023-01-25 at 2 16 10 PM" src="https://user-images.githubusercontent.com/122034315/214560994-d22be53a-fd6a-4f58-9ada-be1ce846e80c.png">

**Comment by nelsonic** [comment]
@seaneady thanks for confirming that you were able to login with `GitHub` 🎉 
the email issue has been logged as: https://github.com/dwyl/auth/issues/264 📧 🙅 🔥

**Comment by nelsonic** [comment]
https://mvp.fly.dev/stats

<img width="1292" alt="image" src="https://user-images.githubusercontent.com/194400/215263563-c7ba0463-d4e5-42cb-8bb3-23fc12ceda66.png">

@seaneady looks like you've been able to authenticate and add a couple of `items`. 👌 
Please confirm if you still see the `Server Error` and if it's no longer an issue, please close. ✅
